get_file.go

```diff
@@ -1,8 +1,27 @@
 package getter
 
+import (
+	"net/url"
+	"os"
+)
+
 // FileGetter is a Getter implementation that will download a module from
 // a file scheme.
 type FileGetter struct {
 	// Copy, if set to true, will copy data instead of using a symlink
 	Copy bool
 }
+
+func (g *FileGetter) ClientMode(u *url.URL) (ClientMode, error) {
+	path := u.Path
+	if u.RawPath != "" {
+		path = u.RawPath
+	}
+
+	// Check if the source is a directory.
+	if fi, err := os.Stat(path); err == nil && fi.IsDir() {
```

Comment:
Should the stat check be extended to whether or not the file exists?

E.g. it looks like the logic says "if I can't stat return ClientModeFile" but this should be an error?

Otherwise seems odd because the validation is asymmetrical. I guess it could be assumed to be checked elsewhere but not clear from this method context.